src/app/pages/checkout/checkout.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+
+import { CheckoutFacade } from 'ish-core/facades/checkout.facade';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CheckoutGuard implements CanActivate {
+  constructor(private checkoutFacade: CheckoutFacade, private router: Router) {}
+
+  canActivate(): boolean | UrlTree {
+    let hasBasket: boolean;
+
+    this.checkoutFacade.basket$.subscribe(value => {
```

Comment:
I have to admit I'm a little surprised this works.  
Subscribing and then comparing the value extracted is not guaranteed to work - imagine if the `basket$` emission is a little delayed for whatever reason. In that case `hasBasket` will not be set correctly even though there might actually be a basket.

In addition, triggering a subscribe on method call without any logic to end the subscription is dangerous. Imagine if I tried to access the checkout route over and over again - every time `canActivate` is called, a new subscription will be opened. There are ways to fix this - check out an example like the [`feature-toggle.directive.ts`](https://github.com/intershop/intershop-pwa/blob/develop/src/app/core/directives/feature-toggle.directive.ts) to see how open subscriptions are canceled. However, this should only really be done in edge cases where nothing else is possible.
[Don't listen to me, listen to the rxjs team lead.](https://benlesh.medium.com/rxjs-dont-unsubscribe-6753ed4fda87).

Thankfully, there is a super simple solution here: [guards can return Observables](https://angular.io/api/router/CanActivate). So just return `this.checkoutFacade.basket$.pipe(....)` and do the value comparison in there.
Angular will take care of subscribing to the observable and using the emitted value for routing - nothing required on your part.

Also, I'm not sure if you need to return `this.router.parseUrl('/')` or if returning `false` would be sufficient.... Maybe you can check.

Reply:
Thanks Max! 
With increasing knowledge the original implementation was now completely overhauled